docs/workshop-hello-world.md

```diff
@@ -0,0 +1,269 @@
+
+- Create 3 sections in the file:
+
+1) A locals block: <https://www.terraform.io/docs/configuration/locals.html> to organize our inputs
+
+```terraform
+locals {
+  my_branch_name = "ibcdpe-1-branch-name"
+  my_namespace_name = "my-cool-namespace"
+  my_application_name_in_argocd = "my-cool-application"
+}
+```
+
+2) A namespace: <https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/>
+
+Update `my-cool-namespace-resource` to whatever value you'd like. In this
+case `my-cool-namespace-resource` is the identifier in which terraform is referencing this specific
+resource.
+
+```terraform
+resource "kubernetes_namespace" "my-cool-namespace-resource" {
+  metadata {
+    name = local.my_namespace_name
+  }
+}
+```
+
+3) An Argo CD Application: <https://argo-cd.readthedocs.io/en/stable/user-guide/application-specification/>
+
+Make sure that `depends_on = [kubernetes_namespace.my-cool-namespace-resource]` is
+referencing the name of the `kubernetes_namespace` resource you just created.
+
+```terraform
+resource "kubectl_manifest" "my-argocd-application" {
+  depends_on = [kubernetes_namespace.my-cool-namespace-resource]
+
+  yaml_body = <<YAML
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: ${local.my_application_name_in_argocd}
+  namespace: argocd
+spec:
+  project: default
+  syncPolicy:
+    automated:
+      prune: true
+  sources:
+  - repoURL: 'https://github.com/Sage-Bionetworks-Workflows/eks-stack.git'
+    targetRevision: ${local.my_branch_name}
+    path: deployments/stacks/dpe-k8s-deployments
+  destination:
+    server: 'https://kubernetes.default.svc'
+    namespace: ${local.my_namespace_name}
+YAML
+}
+```
+
+- Finally let's create a `.yaml` file within `deployments/stacks/dpe-k8s-deployments` that
+defines the kubernetes resources you want to deploy.
+- Create a new file named `deployments.yaml` in the folder `deployments/stacks/dpe-k8s-deployments`
+  - The exact name of the file does not matter, but it should end in `.yaml`
+- In our case we are creating a Deployment: <https://kubernetes.io/docs/concepts/workloads/controllers/deployment/>
+- Add the following to the body of the yaml file:
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: flask-helloworld
+  labels:
+    app: flask-helloworld
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: flask-helloworld
+  template:
+    metadata:
+      labels:
+        app: flask-helloworld
+    spec:
+      containers:
+      - name: flask
+        image: digitalocean/flask-helloworld:latest
+        ports:
+        - containerPort: 5000
+```
+
+- Commit the changes and push the changes to github
+- Create a pull request to main
+- In the git checks that run you'll see a section that says something like: `spacelift/dpe-dev-kubernetes-deployments — 2 to add, 6 to change, 0 to destroy`
+
+![Github check](./workshop-resources/github-check.png)
+
+- Click on `Details` and on the next screen `Deploy`
```

Comment:
I would suggest mentioning that users can `Preview` their app before deploying - I see that's a feature available on the UI:

<img width="379" alt="image" src="https://github.com/user-attachments/assets/dddd6389-33bf-4c13-8f30-c48079b88ed2">

Reply:
Preview doesn't give you anything extra in this case on the Github action. So I was not going to mention it.